src/core/trulens/core/utils/python.py

```diff
@@ -813,18 +817,16 @@ def with_context(context_vars: Optional[ContextVarsOrValues] = None):
             variables to set to their current value.
     """
 
-    tokens = set_context_vars_or_values(context_vars)
-
     try:
+        tokens = set_context_vars_or_values(context_vars)
```

Comment:
TBH, I'm not entirely sure why Piotr put this `token` into the `try` block as I don't think it makes any difference, but I'm keeping it in since it's not harmful or anything and maybe he knew something I didn't.